### PR TITLE
Document the new 9.1 OpenTelemetry meters

### PR DESCRIPTION
**Pull request by mauroservienti**
Documentation for:

- https://github.com/Particular/NServiceBus/pull/7080
- https://github.com/Particular/NServiceBus/pull/7058
- https://github.com/Particular/NServiceBus/pull/7048

### CODE REVIEW

File: nservicebus/operations/opentelemetry_content_core_[8,).partial.md [hunk 1]
```diff
@@ -24,14 +24,22 @@ snippet: opentelemetry-enablemeters
...
 - `nservicebus.messaging.fetches` - Total number of messages fetched from the queue by the endpoint
 - `nservicebus.messaging.failures` - Total number of messages processed unsuccessfully by the endpoint
+- `nservicebus.messaging.handler_time` - The time the user handling code takes to handle a message
+- `nservicebus.messaging.processing_time` - The time the endpoint takes to process a message from when it's fetched from the input queue to when processing completes. It includes:
+  - Invoking all handlers and sagas for a single incoming message
+  - Invoking the incoming message processing pipeline, which includes steps like deserialization or user defined pipeline behaviors.
+- `nservicebus.messaging.critical_time` - The time between when a message is sent and when it is fully processed. It is a combination of:
+  - Network send time: The time a message spends on the network before arriving in the destination queue
+  - Queue wait time: The time a message spends in the destination queue before being picked up and processed
+  - Processing time: The time it takes for the destination endpoint to process the message
```
Comment (mauroservienti) [inline nservicebus/operations/opentelemetry_content_core_[8,).partial.md:34]: What's the best way to make this a partial? Do I need to copy the whole content to a new `nservicebus/operations/opentelemetry_content_core_[9.1,).partial.md` and make the other one `nservicebus/operations/opentelemetry_content_core_[8,9.1).partial.md` or is there a partial of a partial thing?

**Comment by mauroservienti** [comment]
The failure is expected, there isn't a 9.1 NServiceBus version on NuGet yet

File: Snippets/Core/Core_9.1/Core_9.1.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="9.1.*" />
+    <PackageReference Include="NServiceBus.Callbacks" Version="5.*" />
+    <PackageReference Include="NServiceBus.Encryption.MessageProperty" Version="5.*" />
+    <PackageReference Include="NServiceBus.Metrics.ServiceControl" Version="5.*" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.*" />
+    <PackageReference Include="NUnit" Version="3.*" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.17.0" />
+    <PackageReference Include="OpenTelemetry" Version="1.3.0" />
```
Comment (DavidBoike) [inline Snippets/Core/Core_9.1/Core_9.1.csproj:20]: Shouldn't these be safe to remove until they're needed?
```suggestion
```

File: nservicebus/operations/opentelemetry.md [hunk 3]
```diff
@@ -3,10 +3,75 @@ title: OpenTelemetry
...
+
+## Alignment of host identifier
+
+It is recommended to align the instance identifier between NServiceBus and OpenTelemetry so all logs, metrics, traces and audit messages can be correlated by a host (instance) if needed.
+
+> [!NOTE]
+> The OpenTelemetry specification recommends this to be a random uuid. However, it may also be a [deterministic uuid v5](https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/#service-attributes) (i.e. hash of machine name and  endpointname).
+
+NServiceBus adds an [host identifier to all audit messages](/nservicebus/hosting/override-hostid.md) and this instance identifier is also used to show [performance metrics for each running instance in ServicePulse](/monitoring/metrics/in-servicepulse.md). The [instance identifier used for ServicePulse value can be overriden](/monitoring/metrics/install-plugin.md#configuration-instance-id).
```
Comment (DavidBoike) [inline nservicebus/operations/opentelemetry.md:48]:
```suggestion
NServiceBus adds a [host identifier to all audit messages](/nservicebus/hosting/override-hostid.md) and this instance identifier is also used to show [performance metrics for each running instance in ServicePulse](/monitoring/metrics/in-servicepulse.md). The [instance identifier used for ServicePulse value can be overriden](/monitoring/metrics/install-plugin.md#configuration-instance-id).
```

File: nservicebus/operations/opentelemetry.md [hunk 4]
```diff
@@ -3,10 +3,75 @@ title: OpenTelemetry
...
+
+OpenTelemetry also allows to customize the instance id used for `service.instance.id` in various ways.
```
Comment (DavidBoike) [inline nservicebus/operations/opentelemetry.md:50]:
```suggestion
OpenTelemetry also allows customizing the instance id used for `service.instance.id` in various ways.
```

File: nservicebus/operations/opentelemetry.md [hunk 5]
```diff
@@ -3,10 +3,75 @@ title: OpenTelemetry
...
+
+Consider aligning the instance ID used by OpenTelemetry and ServiceControl metrics API.
+
+#### Example
+
+```c#
```
Comment (DavidBoike) [inline nservicebus/operations/opentelemetry.md:56]: Make legit snippet? (Not sure if this would be an 8 and 9 snippet or a 9.1 only snippet)

File: nservicebus/operations/opentelemetry.md [hunk 6]
```diff
@@ -3,10 +3,75 @@ title: OpenTelemetry
...
+// Generate (deterministic) instance ID shared by all components
+
+// Generate deterministic uuid v4 viahttps://github.com/Faithlife/FaithlifeUtility/blob/master/src/Faithlife.Utility/GuidUtility.cs
+var deterministicValue = "MyEndpoint@" + Dns.GetHostName();
+Guid serviceInstanceId = GuidUtility.Create(new Guid("4d63009a-8d0f-11ee-aad7-4c796ed8e320", deterministicValue)) // or Guid.NewGuid()
+
+// OpenTelemetry
+services.AddOpenTelemetry()
+    .ConfigureResource(rb => rb.AddService("MyService", serviceInstanceId: serviceInstanceId.ToString()))
```
Comment (DavidBoike) [inline nservicebus/operations/opentelemetry.md:65]: Guessing `rb` means resource builder? `builder` might be better without being overlong?
```suggestion
    .ConfigureResource(builder => builder.AddService("MyService", serviceInstanceId: serviceInstanceId.ToString()))
```

**Review by DavidBoike — APPROVED** [review]

File: monitoring/metrics/definitions.md [hunk 7]
```diff
@@ -15,42 +15,42 @@ Gathering metrics is important to know how a system works and if it works proper
...
 NServiceBus and ServiceControl capture a number of different metrics about a running endpoint including the processing time, the number of messages in each queue (differentiating between those pulled from the queue, those processed successfully, and those which failed processing), as well as "critical time".
 
-
 ### Processing time
 
-Processing time is the time it takes for an endpoint to **successfully** invoke all handlers and sagas for a single incoming message. Processing failures are not included in the processing time metric.
+Processing time is the time it takes for an endpoint to **successfully** process an incoming message. It includes:
+
+- Invoking all handlers and sagas for a single incoming message
+- Invoking the incoming message processing pipeline, which includes steps like deserialization or user defined pipeline behaviors.
```
Comment (lailabougria) [inline monitoring/metrics/definitions.md:23]:
```suggestion
- The execution of all handlers and sagas for the incoming message
- The execution of the incoming message processing pipeline, which includes deserialization and where applicable, user-defined pipeline behaviors or saga-loading time.
```

File: nservicebus/operations/opentelemetry.md [hunk 8]
```diff
@@ -3,10 +3,75 @@ title: OpenTelemetry
...
+versions: '[8,)'
 related:
  - samples/open-telemetry
 ---
 
 NServiceBus version 8 and above supports [OpenTelemetry](https://opentelemetry.io/docs/instrumentation/net/) through traces, metrics, and logging.
 
-partial: content
\ No newline at end of file
+Enable OpenTelemetry instrumentation in NServiceBus:
```
Comment (lailabougria) [inline nservicebus/operations/opentelemetry.md:13]: Why was this inlined? I just split this up further in another PR that conflicts with this change, so we'll have to coordinate on which structure to keep. Given what I've added now in the tracing section, I think it's getting large enough to benefit from splitting up. See https://github.com/Particular/docs.particular.net/pull/6718

Comment (mauroservienti) [reply nservicebus/operations/opentelemetry.md:13]: Laila, go ahead with that PR. Once that's merged, I'll rebase (or maybe even recreate) this one to align with those larger changes. No worries.

**Comment by SzymonPobiega** [comment]
Closing in favor of https://github.com/Particular/docs.particular.net/pull/6724
